frontend/src/pages/game.tsx

```diff
@@ -0,0 +1,112 @@
+// Game page that is shown when the user goes to localhost:3000/game
+
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { io } from 'socket.io-client';
+import GameControl from '../components/Game/GameControl.tsx';
+
+const Game = () => {
+	const [socket, setSocket] = useState<Socket | null>(null);
+	const [tempToken, setTempToken] = useState<string | null>(null); //tijdelijke oplossing voor Token
+	const [games, setGames] = useState<Match[] | null>([]);
+	const [selectedGame, setSelectedGame] = useState<number | null>(null);
+    
+    useEffect(() => {
+        //because dev mode sometimes didnt disconnect old sockets
+        if (socket) {
+			socket.disconnect(); // Disconnect existing socket if any
+            console.log('Previous socket disconnected');
+        }
+        
+        // Initialize socket connection
+		const token: string = localStorage.getItem('token');
+        const socketIo: Socket = io('http://localhost:3001/game', {
+			transports: ['websocket', 'polling'],
+            query: { token: token } // Hier de token uit localstorage halen
+        });
```

Comment:
Volgens mij heet de token 'authenticationToken' in localstorage, niet 'token', of is dit iets anders?